jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/cpu/OverallCPULoadHandler.java

```diff
@@ -50,16 +58,37 @@ public void initializeMeter(Meter meter) {
             measurement -> {
               measurement.record(machine);
             });
+    meter
+        .gaugeBuilder(METRIC_NAME_MACHINE_MINUTE)
+        .setDescription(METRIC_DESCRIPTION_MACHINE_MINUTE)
+        .setUnit(UNIT_UTILIZATION)
+        .buildWithCallback(
+            measurement -> {
+              measurement.record(machineMinuteAverage);
+            });
   }
 
   @Override
   public void accept(RecordedEvent ev) {
+    // Synchronized to avoid races on machineSum
+    synchronized (this) {
+      if (ev.hasField(MACHINE_TOTAL)) {
+        machine = ev.getDouble(MACHINE_TOTAL);
+        // Remove oldest data point if queue is at capacity
+        if (machineMinuteQueue.size() == 60) {
+          machineSum -= machineMinuteQueue.poll();
+        }
+        // Add new data point
+        machineSum += machine;
+        machineMinuteQueue.add(machine);
+        // Compute new result
+        machineMinuteAverage = machineSum / machineMinuteQueue.size();
```

Comment:
Nice.

Would be good to somehow breakout the pieces of code involved in this so they can be unit tested.

Reply:
I added a [commit](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/651/commits/283ccc38063a7ab6d2c204c6c8bb3ecc33268ee1) that extracts the moving average code and adds a test for it. But I'm not sure it's an ideal solution because the method is now `public`.

Reply:
It can be package private.